File: pointer.js
let address = {
      city: null,
      province: null, 
      country: null
    };
    
    let address1 = {
      ...address,
      city: "Blitar",
      province: "Jawa Timur",
      country: "Indonesia"
    };

    let address2 = {
      ...address1,
      city: "Malang"
    }
    
    console.log(address1)
    console.log(address2)
    

File: pointer.go
package main

import "fmt"

type Address struct {
	City, Province, Country string
}

func main() {
	address1 := Address{"Blitar", "Jawa Timur", "Indonesia"}

	address2 := address1 // Copy Value

	address2.City = "Malang"
	fmt.Println(address1)
	fmt.Println(address2)

	alamat1 := Address{"Blitar", "Jawa Timur", "Indonesia"}

	// alamat2 := &alamat1 // Pointer
	var alamat2 *Address = &alamat1

	alamat2.City = "Malang" // dirubah menjadi malang

	fmt.Println(alamat1) // ikut berubah menjadi malang
	fmt.Println(alamat2) // berubah menjadi malang

}
